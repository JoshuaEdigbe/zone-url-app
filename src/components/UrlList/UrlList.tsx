import React from "react";

// styles
import './index.scss'

interface Props {
  children: React.ReactNode;
}

const UrlList: React.FC<Props> = ({ children }) => {
  return <ul id="url-list-container">{children}</ul>;
};

export default UrlList;
