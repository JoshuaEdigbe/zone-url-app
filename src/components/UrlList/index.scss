#url-list-container {
  list-style: none;
  margin: 0;
  padding: 0;
  background-color: #fff;
  box-shadow: 0px 4px 20px 0px rgba(39, 52, 68, 0.1);
  border-radius: 6px;
  border-top-left-radius: 0;
  border-top-right-radius: 0;


  max-height: 500px;
  overflow-y: scroll;

  li {
    display: flex;
    gap: 20px;
    justify-content: space-between;
    padding: 10px 10px;
    border-bottom: 1px solid #ddd;
    align-items: center;

    .urls-container {
      display: flex;
      flex-direction: column;

      .short-url-text {
        margin-bottom: 3px;
      }

      .old-url-text {
        font-size: 0.8rem;
        color: #5e5e5e;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
      }
    }
    .delete-btn {
      border: 1px solid #ffa7a7;
      border-radius: 3px;
      padding: 0px 7px;
      background-color: #fff;
      height: 26px;
      color: #ffa7a7;
    }
  }
}
