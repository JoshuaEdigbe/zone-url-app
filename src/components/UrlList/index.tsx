
import List from "./UrlList";
import Item from "./_partials/UrlListItem";

export const UrlList = {
  List,
  Item,
};
