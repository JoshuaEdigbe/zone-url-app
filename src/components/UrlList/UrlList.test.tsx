import React from "react";
import { render } from "@testing-library/react";
import UrlList from "./UrlList";

describe("UrlList component", () => {
  it("renders without crashing", () => {
    render(<UrlList>Test children</UrlList>);
  });

  it("renders children correctly", () => {
    const { getByText } = render(<UrlList>Test children</UrlList>);
    const childrenElement = getByText("Test children");

    expect(childrenElement).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<UrlList>Test children</UrlList>);
    expect(container).toMatchSnapshot();
  });
});
