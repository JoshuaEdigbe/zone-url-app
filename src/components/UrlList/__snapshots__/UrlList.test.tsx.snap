// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlList component matches snapshot 1`] = `
<div>
  <ul
    id="url-list-container"
  >
    Test children
  </ul>
</div>
`;
